src/app/settings/shelf-settings/components/edit-shelf-action-panel-configuration/edit-shelf-action-panel-configuration.component.ts

```diff
@@ -51,4 +55,37 @@ export class EditShelfActionPanelConfigurationComponent implements OnInit {
   public cancel(): void {
     this.onCancel.emit()
   }
+
+  public get saveButtonErrorsTooltip(): string | undefined {
+    const invalidControlsLabels: string[] = this.findInvalidControlsLabels()
+    return invalidControlsLabels.length > 0 ? $localize`common.form-required-field.button-tooltip` + `: ${invalidControlsLabels.join(', ')}` : undefined
+  }
+
+  private findInvalidControlsLabels(): string[] {
+    const invalidControlLabels: string[] = []
+    const controls = this.form.controls
+    for (const name in this.form.controls) {
+      if (controls[name].valid) {
+        continue
+      }
+      const translateControl: string | undefined = this.translateControl(name)
+      if (translateControl) {
+        invalidControlLabels.push(translateControl)
+      }
+    }
+    return invalidControlLabels
+  }
+
+  private translateControl(controlName: string): string | undefined {
+    switch (controlName) {
+      case FORM_NAME_CONTROL_LABEL:
+        return $localize`action-panel.panel-name.label`
+      case FORM_ACTION_FILTER_CONTROL_LABEL:
+        return $localize`global.filters.label`
+      case FORM_RANK_CONTROL_LABEL:
+        return $localize`action-panel.rank.label`
+      default:
+        return undefined
```

Comment:
Same comment about returning the control name instead of undefined when translating.

Reply:
@LindvedKrvang when any of the controls in `actionTriggerDataForm ` has a error data control in `actionTriggerForm` has a error too. That way we return only controls witch want to be shown in case in data control (witch is control group) we return undefined and we skip it. If we want to rework that we need to find another solution for that row:
`const controls = { ...this.actionTriggerForm.controls, ...this.actionTriggerDataForm.controls }`

Reply:
@LindvedKrvang I filter the `data` control and that fix the issue.